gateway/gateway.go

```diff
@@ -515,10 +518,10 @@ func (gateway *HandleT) userWebRequestWorkerProcess(userWebRequestWorker *userWe
 				userIDFromReq := strings.TrimSpace(vjson.Get("userId").String())
 				if builtUserID == "" {
 					if anonIDFromReq != "" {
-						builtUserID = anonIDFromReq + DELIMITER + userIDFromReq
+						builtUserID = userIDHeader + DELIMITER + anonIDFromReq + DELIMITER + userIDFromReq
 					} else {
 						// Proxy gets the userID from body if there is no anonymousId in body/header
-						builtUserID = userIDFromReq + DELIMITER + userIDFromReq
+						builtUserID = userIDHeader + DELIMITER + userIDFromReq + DELIMITER + userIDFromReq
 					}
```

Comment:
How much effort would it require to write tests for this section? Maybe isolating this to a function would make it possible or easier.

Reply:
We don't have such tests for gateway :/ I can isolate this part and add it though if it makes sense

Reply:
Can't we copy from `multi_tenant_test.go` and have the app start in GATEWAY mode only, send an event with the correct header and then check the `user_id` on the DB? Or you think it's going to be more complex?